hooks/useSingleRecord.ts

```diff
@@ -30,8 +29,19 @@ const useSingleRecord = () => {
   const id: string = requestParams[1]
 
   useEffect(() => {
-    if (requestReady && id !== '[id]')
-      fetchSingleOrgRecord(id, setSingleFetchedRecord)
+    const airtableApiRouteFetch = async () => {
+      if (requestReady && id !== '[id]') {
+        const apiRequest = await fetch('/api/singleairtablerecord', {
+          method: 'POST',
+          body: JSON.stringify({ id }),
+        })
+
+        const apiResponse = await apiRequest.json()
+
+        setSingleFetchedRecord(apiResponse)
+      }
+    }
+    airtableApiRouteFetch()
```

Comment:
what was going on with lines 33-34

Reply:
@vsauceda-cfa do you mean the check where i'm making sure that it doesn't equal `[id]`? I'm doing that because the first time the component renders, it will feed the hook info from the router. the routers id is also the record number for the org to be fetched. before it is ready, the router will read the address as '[id]` to show that it is a dynamic route that will _eventually_ be an id. after the id number loads, that string becomes something else, and then it is ready to fetch the record from airtable.

huh, that's a bit of a roundabout explanation. does that make sense?